modules/transactions/src/main/java/org/apache/ignite/internal/tx/impl/TxManagerImpl.java

```diff
@@ -530,7 +530,7 @@ private CompletableFuture<Void> makeFinishRequest(
                 commitPartition, primaryConsistentId, term, commit, txId, replicationGroupIds);
 
         return txMessageSender.finish(primaryConsistentId, commitPartition, replicationGroupIds, txId, term, commit, commitTimestamp)
-                .thenRun(() -> {
+                .thenAccept(txResult -> {
                     updateTxMeta(txId, old -> {
```

Comment:
Might we complete txFinishFuture out of the update closuer to avoid invoking that undr a lock?